diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0fc9e2cfd..3bab94cbe 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -115,7 +115,7 @@ endif(compiler_only OR build_all)
 
 # Find required dependencies for thrift/lib
 if(lib_only OR build_all)
-  find_package(Gflags REQUIRED)
+  find_package(gflags CONFIG QUIET)
   find_package(Glog REQUIRED)
   find_package(folly CONFIG REQUIRED)
   find_package(yarpl CONFIG REQUIRED)
diff --git a/thrift/lib/cpp/CMakeLists.txt b/thrift/lib/cpp/CMakeLists.txt
index a0dbefbf2..5076a108c 100644
--- a/thrift/lib/cpp/CMakeLists.txt
+++ b/thrift/lib/cpp/CMakeLists.txt
@@ -46,8 +46,9 @@ target_link_libraries(
   concurrency
   PUBLIC
     Folly::folly
-    ${LIBGFLAGS_LIBRARY}
     ${GLOG_LIBRARIES}
+  PRIVATE
+    gflags
 )
 
 bypass_source_check("${reflection-cpp2-SOURCES}")
diff --git a/thrift/lib/cpp2/CMakeLists.txt b/thrift/lib/cpp2/CMakeLists.txt
index 00b5e161b..f401dc228 100644
--- a/thrift/lib/cpp2/CMakeLists.txt
+++ b/thrift/lib/cpp2/CMakeLists.txt
@@ -60,7 +60,8 @@ target_link_libraries(
     Folly::folly
     thriftmetadata
     ${GLOG_LIBRARIES}
-    ${LIBGFLAGS_LIBRARY}
+  PRIVATE
+    gflags
 )
 
 add_library(
